src/test/scala/fm/http/TestClientAndServer.scala

```diff
@@ -357,6 +364,12 @@ final class TestClientAndServer extends FunSuite with Matchers with BeforeAndAft
     getSync("/redirect6", 302, "/redirect5", clientNoFollowRedirects)
   }
 
+  test("Authentication - Basic Auth") {
+    getSync("/basic_auth", 401, "You need a valid user and password to access this content.")
+    getSync("/basic_auth", 401, "You need a valid user and password to access this content.", client.withBasicAuth("bar", "foo"))
+    getSync("/basic_auth", 200, "ok", client.withBasicAuth("foo", "bar"))
+  }
```

Comment:
This would be more useful if the server side piece of this made use of `fm.http.server.BasicAuth` to test the server side implementation instead of the manually coded `handleBasicAuth` method.

Reply:
I looked into it, but it wasn't obvious to me howto do something quickly.  Feel free to fix it up.  The client header was the more important bug I was trying to verify/fix for `ta-webservice-api`